ocrd_models/ocrd_page_user_methods/get_AllTextLines.py

```diff
@@ -0,0 +1,19 @@
+def get_AllTextLines(self, region_order='document', textline_order='top-to-bottom'):
+    """
+    Return all the TextLine in the document
+
+    Arguments:
+        region_order ("document"|"reading-order"|"reading-order-only") Whether to
+            return regions sorted by document order (``document``, default) or by
+            reading order with regions not in the reading order at the end of the
+            returned list (``reading-order``) or regions not in the reading order
+            omitted (``reading-order-only``)
+        textline_order ("top-to-bottom"|"bottom-to-top"|left-to-right"|"right-to-left")
+            The order of text lines within a block (not currently used)
+    """
+    # TODO handle textLineOrder
```

Comment:
I think the mnemonics should be different, `textline_order` should be a boolean.

If true, then do something like:
```python
    for reg in self.get_AllRegions(['Text'], order=region_order):
        lo = reg.get_textLineOrder() or self.get_textLineOrder() or 'top-to-bottom'
        lines = reg.get_TextLine()
        ret += lines if lo in ['top-to-bottom', 'left-to-right'] else list(reversed(lines))
    return ret
```

Not entirely sure about the PAGE-XML semantics here TBH. Must XML ordering always be top-to-bottom (or left-to-right in case of vertical text)? In that case, you would need to _apply_ `@textLineOrder` to get the correct text concatenation (`right-to-left` for classical Chinese). But the attribute could also be interpreted to _describe_ just how the current XML ordering looks like (which you could verify by its coordinates). In that case, you would never re-order anything.

Reply:
Thanks, I'll implement it like that for now until we agreed on the right interpretation in https://github.com/PRImA-Research-Lab/PAGE-XML/issues/26

Reply:
012e49ea The kwarg is now `respect_textline_order` and the algorithm like you proposed.